commerce/domain/pricing/cart/calculator/EstimatedActualPricingCalculator.cls

```diff
@@ -0,0 +1,229 @@
+                                        cartItem);
+                String errorMessage = getFailedToRepriceItemMessage(cartItem);
+                cvo.setMessage(errorMessage);
+                cart.getCartValidationOutputs().add(cvo);
+                continue;
+            }
+            Decimal price = pricingDataMap.get(cartItem.getSku());
+            // Update cart item fields
+            cartItem.setListPrice(price);
+            cartItem.setSalesPrice(price);
+            cartItem.setTotalListPrice(price * cartItem.getQuantity());
+            cartItem.setTotalPrice(price * cartItem.getQuantity());
+        }
+    }
+
+    /**
+     * @description Returns an iterator instance for Cart Items that need price calculations.
+     *
+     * @param cart In memory representation of the Cart
+     * @param optionalBuyerActionDetails The latest set of changes applied to the Cart by the Buyer
+     *
+     * @return <<Iterator<CartExtension.CartItem>>>
+     */
+    private Iterator<CartExtension.CartItem> clearErrorsAndGetCartItemsIterator(
+            CartExtension.Cart cart,
+            CartExtension.OptionalBuyerActionDetails optionalBuyerActionDetails) {
+
+        CartExtension.CartValidationOutputList cartValidationOutputs = cart.getCartValidationOutputs();
+        if (!optionalBuyerActionDetails.isPresent() || optionalBuyerActionDetails.get().isCheckoutStarted()) {
+            // Remove all CVOs of PRICING type since we are going to reprice all cart items.
+            removeAllCVOsOfType(cart, CartExtension.CartValidationOutputTypeEnum.PRICING);
+            return cart.getCartItems().iterator();
+        }
+
+        // Get the list of changed cart items from BuyerActionDetails
+        List<CartExtension.CartItem> cartItemsToCalculatePrice = new List<CartExtension.CartItem>();
+        List<CartExtension.CartItemChange> cartItemChanges = optionalBuyerActionDetails.get().getCartItemChanges();
+
+        for (CartExtension.CartItemChange cartItemChange : cartItemChanges) {
+            CartExtension.OptionalCartItem optionalCartItem = cartItemChange.getChangedItem();
+            if (optionalCartItem.isPresent()) {
+                CartExtension.CartItem cartItem = optionalCartItem.get();
+                cartItemsToCalculatePrice.add(cartItem);
+                // Remove CVOs related to the changed cart item, but leave other PRICING related CVOs.
+                removeCVOsOfTypeRelatedTo(cart, CartExtension.CartValidationOutputTypeEnum.PRICING, cartItem);
+            }
+        }
+
+        return cartItemsToCalculatePrice.iterator();
+    }
+
+    private String getGenericErrorMessage() {
+        // Use User locale to populate localized message.
+        if (UserInfo.getLocale() == 'fr') {
+            return GENERAL_ERROR_MESSAGE_FR;
+        } else {
+            return GENERAL_ERROR_MESSAGE_EN;
+        }
+        // Alternatively Labels can be used.
+        //return Label.GENERIC_PRICING_FAILED_ERROR_MESSAGE;
```

Comment:
Unfortunately Salesforce does not allow Label creation in scope of Apex test. That's why I don't provide Label as an main example but keep it commented out.